src/main/java/Util/Ui.java

```diff
@@ -0,0 +1,244 @@
+package Util;
+
+import Tasks.Deadline;
+import Tasks.Event;
+import Tasks.Task;
+import Tasks.Todo;
+import Tasks.TaskList;
+
+import java.text.DateFormat;
+import java.text.DateFormatSymbols;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Scanner;
+
+import Exception.DukeException;
+import Exception.IncorrectTaskNameException;
+import Exception.EmptyDeadlineDescriptionException;
+import Exception.EmptyEventDescriptionException;
+import Exception.EmptyTodoDescriptionException;
+
+/**
+ * Handles the User Interface for the program where the user will interact with
+ * by inputting data into the console. The class will then take this input and
+ * interact with the associated TaskList and Storage classes
+ */
+public class Ui {
+
+    private Parser parser;
+    private TaskList tasksList;
+    private Storage storage;
+
+    /**
+     * To store the several suffixed that will be used in a month starting from
+     * 0 ending at 31. (i.e 2nd, 11th, 23rd)
+     */
+    static String[] suffixes =
+            { "th", "st", "nd", "rd", "th", "th", "th", "th", "th", "th",
+                    "th", "th", "th", "th", "th", "th", "th", "th", "th", "th",
+                    "th", "st", "nd", "rd", "th", "th", "th", "th", "th", "th",
+                    "th", "st" };
+
+    public Ui(TaskList taskList, Storage storage) {
+        this.tasksList = taskList;
+        this.storage = storage;
+    }
+
+    /**
+     * take in input from user and then perform relevant methods to read/write
+     * into local storage and save into TaskList
+     * Takes in only the follow inputs (bye, list, done, delete, todo, deadline, event)
+     */
+    public void readInput() throws Exception, DukeException {
+        Scanner scanner = new Scanner(System.in);
+        showWelcome();
+        while(scanner.hasNextLine()) {
+            String input = scanner.nextLine();
+            if(input.equals("bye")) {
+                showExit();
+                break;
+            }
+            if(input.equals("list")) {
+                tasksList.showAllTasks();
+                continue;
+            }
+            if((input.length() > 5) && input.substring(0, 5).equals("done ")) {
+                showDone(input);
+                continue;
+            }
+            if((input.length() > 7) && input.substring(0, 7).equals("delete ")) {
+                showDelete(input);
+                continue;
+            }
+            if((input.length() > 5) && input.substring(0, 5).equals("find ")) {
+                showFind(input);
+                continue;
+            }
+            else {
+                // just to catch wrong input
+                try {
+                    input.substring(0,4).equals("todo");
```

Comment:
Should just add it to the bottom of the switch, anything you don't recognize is bad input and you don't need to check for it explicitly.